crates/rethnet_rpc_server/tests/integration_tests.rs

```diff
@@ -58,163 +63,124 @@ async fn submit_request(address: &SocketAddr, request: &RpcRequest<MethodInvocat
         .unwrap_or_else(|_| panic!("should get full response text"))
 }
 
-#[tokio::test]
-async fn test_get_balance_nonexistent_account() {
+async fn verify_response<ResponseT>(
+    server: &SocketAddr,
+    method: MethodInvocation,
+    response_data: jsonrpc::ResponseData<ResponseT>,
```

Comment:
Is this ever an error response? If not, you can just construct the success internally from the `ResponseT`, saving you some more code duplication in your tests

Reply:
great suggestion, thanks, addressed in https://github.com/NomicFoundation/hardhat/pull/4132/commits/f354e9d10f1faa268848dcf007749c47dace43ef